include/verilated_types.h

```diff
@@ -601,14 +623,21 @@ template <class T_Key, class T_Value> class VlAssocArray final {
     T_Value& at(const T_Key& index) {
         const auto it = m_map.find(index);
         if (it == m_map.end()) {
-            std::pair<typename Map::iterator, bool> pit = m_map.emplace(index, m_defaultValue);
-            return pit.first->second;
+            modified = true;
+            auto it2 = std::find_if(modify_input.begin(), modify_input.end(), [&](const auto& pair) {
+                return pair.first == index;
+            });
+            if (it2 == modify_input.end()) {
+                modify_input.emplace_back(index, m_defaultValue);
+                return modify_input.back().second;
+            }
+            return (*it2).second;
         }
         return it->second;
     }
```

Comment:
Given that we need to do a `m_map.find` in all cases, I think this would be faster (and simpler) if we simply used:
```
    T_Value& at(const T_Key& index) {
        return m_map.emplace(index, m_defaultValue).first->second;
    }
```

This works because if the element already exists, emplace does not change it but returns an iterator to it. Otherwise it constructs it, in the place that it already knows it belongs to, so you save a search. Please check if this would be faster.

Reply:
> Given that we need to do a `m_map.find` in all cases, I think this would be faster (and simpler) if we simply used:
> 
> ```
>     T_Value& at(const T_Key& index) {
>         return m_map.emplace(index, m_defaultValue).first->second;
>     }
> ```
> 
> This works because if the element already exists, emplace does not change it but returns an iterator to it. Otherwise it constructs it, in the place that it already knows it belongs to, so you save a search. Please check if this would be faster.

All requirements have been done except this,
Maybe I misunderstand your meaning, but I think it is hard to archive.
In the case, the key/value doesn't exist in m_map, but it exists in m_modify_input
emplace will make a new entry

Reply:
I will comment on the new PR #3342